first-release (#119)
iter8-tools/docs: docs/getting-started/first-abn.md

--8<-- "docs/getting-started/install.md"

```shell
export IMG=kalantar/iter8:20231011-1506
export CHARTS=/Users/kalantar/projects/go.workspace/src/github.com/iter8-tools/iter8/charts
helm upgrade --install iter8 $CHARTS/controller \
--set image=$IMG --set logLevel=trace \

> --set clusterScoped=true

Reviewer: Nicer to see all the parameter on their own line. I'm wondering if it's necessary to keep the `logLevel`? In any case, I like to keep the least important parameter at the end.

> --set clusterScoped=true

Author: This was temporary for testing. Removed

    The frontend component is implemented to call `Lookup()` before each call to the backend component. The frontend component uses the returned version number to route the request to the recommended version of the backend component.

=== "backend"

>     Release and initial version of the backend named `backend`:

Reviewer: ```suggestion
    Release an initial version of the backend named `backend`:
```

## Deploy candidate

> Release the application adding a second (candidate) version named `backend-candidate-1`:

Reviewer: I think I still prefer the original wording. The new wording is less clear to me

```shell
cat <<EOF | helm upgrade --install backend $CHARTS/release -f -
environment: deployment
application: 
  port: 8091
  versions:
  - metadata:
      name: backend
    image: iter8/abn-sample-backend:0.17-v1
  - metadata:
      name: backend-candidate-1
    image: iter8/abn-sample-backend:0.17-v2
EOF
```

> While the candidate version is deploying, `Lookup()` will continue to return just the primary version. Once the candidate is ready, it will begin recommending both versions allowing traffic to be distributed across both.z

Reviewer: ```suggestion
While the candidate version is deploying, `Lookup()` will only return the primary version. Once the candidate is ready, it will begin recommending both versions, allowing traffic to be distributed across both.
```

Reviewer: I think it should be mentioned here that the `versions` list is ordered and the first version is the primary version.

The Iter8 dashboard allows you to compare the behavior of the two versions of the backend component against each other and select a winner. Since user requests are being sent by the load generation script, the values in the report may change over time. The Iter8 dashboard will look like the following:

![A/B dashboard](../tutorials/images/abnDashboard.png)

Once you identify a winner, it can be promoted, and the candidate version deleted.

## Promote candidate

> To promote the candidate version (`backend-candidate-1`), re-release the application, updating the image of the first (primary) version and remove the candidate version:

Reviewer: ```suggestion
To promote the candidate version (`backend-candidate-1`), re-release the application, updating the image of the primary version (the first version) to use the image of the candidate version and remove the old candidate version:
```

```shell
cat <<EOF | helm upgrade --install backend $CHARTS/release -f -
environment: deployment
application: 
  port: 8091
  versions:
  - metadata:
      name: backend
    image: iter8/abn-sample-backend:0.17-v2
EOF
```

> Calls to `Lookup()` will now recommend that all traffic be sent to the primary version `backend` (currently serving the promoted version of the code).

Reviewer: ```suggestion
Calls to `Lookup()` will now recommend that all traffic be sent to the new primary version `backend` (currently serving the promoted version of the code).
```
